src/ssh.c

```diff
@@ -205,8 +219,13 @@ int dcc_ssh_connect(char *ssh_cmd,
     if (!path)
         path = (char *) "distccd";
 
+    char *child_argv[10+num_ssh_args];
```

Comment:
10 looks like a magic number, possibly relating to the value found in: 
`char *child_argv[10];` 
However this was replaced with `char *ssh_args[6];` in your patch.  
Also could the variable declaration be moved to the start of the function please?